Applications/Utils/MeshEdit/AddTopLayer.cpp

```diff
@@ -0,0 +1,175 @@
+/*
+ * \date 2015-04-14
+ * \brief Adds a top layer to an existing mesh.
+ */
+
+// TCLAP
+#include "tclap/CmdLine.h"
+
+// ThirdParty/logog
+#include "logog/include/logog.hpp"
+
+// BaseLib
+#include "BaseLib/LogogSimpleFormatter.h"
+#include "BaseLib/FileTools.h"
+
+// FileIO
+#include "FileIO/readMeshFromFile.h"
+#include "FileIO/XmlIO/Boost/BoostXmlGmlInterface.h"
+#include "FileIO/VtkIO/VtuInterface.h"
+
+// GeoLib
+#include "GeoLib/PointWithID.h"
+
+// MeshLib
+#include "MeshLib/Mesh.h"
+#include "MeshLib/Node.h"
+#include "MeshLib/Elements/Element.h"
+#include "MeshLib/Elements/Prism.h"
+#include "MeshLib/Elements/Hex.h"
+#include "MeshLib/Elements/Quad.h"
+
+// MeshGeoToolsLib
+#include "MeshGeoToolsLib/MeshNodeSearcher.h"
+
+int main (int argc, char* argv[])
+{
+	LOGOG_INITIALIZE();
+	logog::Cout* logog_cout (new logog::Cout);
+	BaseLib::LogogSimpleFormatter *custom_format (new BaseLib::LogogSimpleFormatter);
+	logog_cout->SetFormatter(*custom_format);
+
+	TCLAP::CmdLine cmd(
+		"Adds a top layer to an existing 3d mesh",
+		' ',
+		"0.1");
+
+	TCLAP::ValueArg<std::string> mesh_arg("m", "mesh-file",
+		"the name of the file containing the mesh", true,
+		"", "file name");
+	cmd.add(mesh_arg);
+
+	TCLAP::ValueArg<std::string> mesh_out_arg("o", "output-mesh-file",
+		"the name of the file the mesh should be written to (vtu format)", true,
+		"", "file name");
+	cmd.add(mesh_out_arg);
+
+	TCLAP::ValueArg<double> layer_thickness_arg("t", "layer-tickness",
+		"the thickness of the new layer", false, 10, "floating point value");
+	cmd.add(layer_thickness_arg);
+
+	cmd.parse(argc, argv);
+
+	INFO("Reading mesh \"%s\" ... ", mesh_arg.getValue().c_str());
+	MeshLib::Mesh * subsfc_mesh(FileIO::readMeshFromFile(mesh_arg.getValue()));
+	INFO("done.");
+
+	INFO("Extracting top surface of mesh \"%s\" ... ",
+		mesh_arg.getValue().c_str());
+	const MathLib::Vector3 dir(0,0,-1);
+	double const angle(90);
+	std::unique_ptr<MeshLib::Mesh> sfc_mesh(
+		MeshLib::MeshSurfaceExtraction::getMeshSurface(
+			*subsfc_mesh, dir, angle, true
+		)
+	);
+	INFO("done.");
```

Comment:
you are not testing if the input mesh is actually 3D (not that it matters, the method will work just as well with a 2D mesh ... it is just not consistent with the documentation in the beginning)